lib/booking_manager.rb

```diff
@@ -0,0 +1,59 @@
+require_relative 'reservation.rb'
+require 'pry'
+
+class BookingManager
+
+  attr_reader :rooms
+  attr_accessor :reservations, :start_date, :end_date, :make_reservation
+
+
+  def initialize
+    @start_date = start_date
+    @end_date = end_date
+    @reservations = []
+    @rooms = []
+    i = 0
+    20.times do |i|
+      @rooms << i + 1
+      i += 1
+    end
+
+  end
+
+  def find_available_room(start_date, end_date)
+    requested_dates = (start_date...end_date).to_a
+    room = 237
+    room_count = 1
+    while room_count < 20
+      if @reservations.empty?
+        room = 1
+        return room
+      end
+      @reservations.each_with_index do |reservation, index|
+        if reservation.room == (index + 1) && (reservation.reservation_dates & requested_dates).length == 0
+          room = index + 1
+          return room
+        elsif reservation.room == (index + 1)
+          room_count += 1
+        elsif reservation.room != (index + 1)
+          room = index + 1
+          return room
+        end
+      end
```

Comment:
Overall this `if ... elsif ... elsif` block feels pretty unclear to me about what's going on -- maybe making tests, helper methods, adding comments, or trying a different structure between `if ... elsif ... elsif` would help